src/metalperformanceshaders.cs

```diff
@@ -3340,91 +4611,526 @@ interface MPSNNBinaryArithmeticNode {
+
+		[Export ("scaleFactorX")]
+		double ScaleFactorX { get; }
+
+		[Export ("scaleFactorY")]
+		double ScaleFactorY { get; }
+	}
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[BaseType (typeof (MPSNNGradientFilterNode), Name = "MPSCNNUpsamplingBilinearGradientNode")]
+	[DisableDefaultCtor]
+	interface MPSCnnUpsamplingBilinearGradientNode {
+
+		[Static]
+		[Export ("nodeWithSourceGradient:sourceImage:gradientState:scaleFactorX:scaleFactorY:")]
+		MPSCnnUpsamplingBilinearGradientNode NodeWithSourceGradient (MPSNNImageNode sourceGradient, MPSNNImageNode sourceImage, MPSNNGradientStateNode gradientState, double scaleFactorX, double scaleFactorY);
+
+		[Export ("initWithSourceGradient:sourceImage:gradientState:scaleFactorX:scaleFactorY:")]
+		IntPtr Constructor (MPSNNImageNode sourceGradient, MPSNNImageNode sourceImage, MPSNNGradientStateNode gradientState, double scaleFactorX, double scaleFactorY);
+
+		[Export ("scaleFactorX")]
+		double ScaleFactorX { get; }
+
+		[Export ("scaleFactorY")]
+		double ScaleFactorY { get; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSKernel))]
+	[DisableDefaultCtor] // There is a DesignatedInitializer, file a bug if needed.
+	interface MPSNNGraph : NSCopying, NSSecureCoding {
+
+		[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+		[Export ("initWithDevice:resultImage:resultImageIsNeeded:")]
 		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, MPSNNImageNode resultImage, bool resultIsNeeded);
+
+		[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+		[Static]
+		[Export ("graphWithDevice:resultImage:resultImageIsNeeded:")]
+		[return: NullAllowed]
+		MPSNNGraph Create (IMTLDevice device, MPSNNImageNode resultImage, bool resultIsNeeded);
+
+		[Deprecated (PlatformName.TvOS, 11, 3, message: "Use '.ctor (IMTLDevice, MPSNNImageNode, bool)' instead.")]
+		[Deprecated (PlatformName.iOS, 11, 3, message: "Use '.ctor (IMTLDevice, MPSNNImageNode, bool)' instead.")]
+		[Deprecated (PlatformName.MacOSX, 10, 13, 4, message: "Use '.ctor (IMTLDevice, MPSNNImageNode, bool)' instead.")]
+		[Export ("initWithDevice:resultImage:")]
 		IntPtr Constructor (IMTLDevice device, MPSNNImageNode resultImage);
 
+		// TODO: Review in Xcode 11 / iOS 13
+		// Looks like the following two were exposed ahead of their time
+		// [TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+		// [Export ("initWithDevice:resultImages:resultsAreNeeded:")]
+		// [DesignatedInitializer]
+		// IntPtr Constructor (MTLDevice device, MPSNNImageNode[] resultImages, [NullAllowed] bool* areResultsNeeded);
+
+		// [TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+		// [Static]
+		// [Export ("graphWithDevice:resultImages:resultsAreNeeded:")]
+		// [return: NullAllowed]
+		// MPSNNGraph GraphWithDevice (MTLDevice device, MPSNNImageNode[] resultImages, [NullAllowed] bool* areResultsNeeded);
```

Comment:
^ ???

Reply:
Yes those two seemed to be leaked into this beta release so just a heads up for our future selves